ank/src/filtered_complete_state.rs

```diff
@@ -0,0 +1,138 @@
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use api::ank_base;
+use common::{
+    helpers::serialize_to_ordered_map,
+    objects::{AddCondition, ControlInterfaceAccess, RestartPolicy, Tag, WorkloadStatesMap},
+};
+use serde::{Deserialize, Serialize, Serializer};
+
+use crate::output_and_error;
+
+pub fn serialize_option_to_ordered_map<S, T: Serialize>(
+    value: &Option<HashMap<String, T>>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    if let Some(value) = value {
+        serialize_to_ordered_map(value, serializer)
+    } else {
+        serializer.serialize_none()
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FilteredCompleteState {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub desired_state: Option<FilteredState>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub workload_states: Option<WorkloadStatesMap>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FilteredState {
+    // [impl->swdd~cli-returns-api-version-with-desired-state~1]
+    pub api_version: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default, serialize_with = "serialize_option_to_ordered_map")]
+    pub workloads: Option<HashMap<String, FilteredWorkloadSpec>>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct FilteredWorkloadSpec {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub agent: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tags: Option<Vec<Tag>>,
+    #[serde(serialize_with = "serialize_option_to_ordered_map")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dependencies: Option<HashMap<String, AddCondition>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub restart_policy: Option<RestartPolicy>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub runtime: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub runtime_config: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub control_interface_access: Option<ControlInterfaceAccess>,
+}
+
+impl From<ank_base::CompleteState> for FilteredCompleteState {
+    fn from(value: ank_base::CompleteState) -> Self {
+        FilteredCompleteState {
+            desired_state: value.desired_state.map(|x| x.into()),
+            workload_states: value.workload_states.map(|x| x.into()),
```

Comment:
You do not need anonymous functions here: 
```suggestion
            desired_state: value.desired_state.map(Into::into),
            workload_states: value.workload_states.map(Into::into),
```